### PR TITLE
Allow primary keys with `auto_increment` columns to be dropped when an appropriate index exists

### PR DESCRIPTION
**Pull request by fulghum**
MySQL allows a primary key with an `auto_increment` column to be dropped as long as there is a secondary index that includes the `auto_increment` column as the first column in the index. ([MySQL Reference](https://dev.mysql.com/doc/refman/8.0/en/innodb-auto-increment-handling.html))

This PR also enables dropping a primary key by referencing it by it's ID (`PRIMARY`), in order to match MySQL's behavior, e.g. 
```
DROP INDEX `PRIMARY` ON t;
```

Related to https://github.com/dolthub/dolt/issues/7456

### CODE REVIEW

File: enginetest/queries/alter_table_queries.go [hunk 1]
```diff
@@ -1331,6 +1331,139 @@ var AddDropPrimaryKeyScripts = []ScriptTest{
...
+		// referring to it with the 'PRIMARY' name.
+		Name: "Drop auto-increment primary key as named index",
+		SetUpScript: []string{
+			"create table t (id int primary key AUTO_INCREMENT, c1 varchar(255));",
+			"insert into t (c1) values ('one');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				// Without a supporting index, we should get an error
+				Query:       "DROP INDEX `PRIMARY` ON t;",
```
Comment (zachmu) [inline enginetest/queries/alter_table_queries.go:1345]: Do we even support this syntax? Isn't it `drop primary key?`

File: enginetest/queries/alter_table_queries.go [hunk 2]
```diff
@@ -1331,6 +1331,139 @@ var AddDropPrimaryKeyScripts = []ScriptTest{
...
+			{
+				Query:    "DROP INDEX `PRIMARY` ON t;",
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query:    "show create table t;",
+				Expected: []sql.Row{{"t", "CREATE TABLE `t` (\n  `id` int NOT NULL AUTO_INCREMENT,\n  `c1` varchar(255),\n  UNIQUE KEY `id` (`id`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin"}},
+			},
+			{
+				Query:    "SELECT * FROM t;",
```
Comment (zachmu) [inline enginetest/queries/alter_table_queries.go:1361]: Might do an insert here to make sure the auto inc key still works

File: enginetest/queries/alter_table_queries.go (hunk 1 above)
Comment (zachmu) [reply enginetest/queries/alter_table_queries.go:1345]: Guess it works in MySQL!

**Review by zachmu — APPROVED** [review]
LGTM!

File: enginetest/queries/alter_table_queries.go (hunk 2 above)
Comment (fulghum) [reply enginetest/queries/alter_table_queries.go]: Good suggestion. This turned up some problems in the Dolt layer! 🙏

**Pull request merged**